src/test/java/org/killbill/billing/plugin/vertex/VertexTaxCalculatorTest.java

```diff
@@ -231,6 +231,95 @@ public void testTaxDescription() throws Exception {
+        final TaxesType taxesType = new TaxesType();
+        taxesType.setCalculatedTax(MOCK_TAX_AMOUNT_1_01);
+        given(responseLineItem.getTaxes()).willReturn(Collections.singletonList(taxesType));
+
+        //given tax effective rate is present
+        taxesType.setEffectiveRate(0.09975d);
+
+        //then it persisted in item details invoice item field
+        List<InvoiceItem> result = vertexTaxCalculator.compute(account, invoice, true, Collections.emptyList(), tenantContext);
+        assertEquals("{\"taxRate\":0.09975}", result.get(0).getItemDetails());
+        checkTaxItemFields(result.get(0));
+
+        //given effective rate is not present
+        taxesType.setEffectiveRate(null);
+
+        //then no item details persisted
+        result = vertexTaxCalculator.compute(account, invoice, true, Collections.emptyList(), tenantContext);
+        assertNull(result.get(0).getItemDetails());
+        checkTaxItemFields(result.get(0));
+    }
+
+    @Test(groups = "fast")
+    public void testComputeWhenTaxEffectiveRateIsNotPresented() throws Exception {
+        //given
+        given(invoice.getInvoiceItems()).willReturn(Collections.singletonList(taxableInvoiceItem));
+        given(taxResponse.getData()).willReturn(apiResponseData);
+        final TaxesType taxesType = new TaxesType();
+        taxesType.setCalculatedTax(MOCK_TAX_AMOUNT_1_01);
+        given(responseLineItem.getTaxes()).willReturn(Collections.singletonList(taxesType));
+
+        //given tax effective rate is not present
+        taxesType.setEffectiveRate(null);
+
+        //then it persisted in item details invoice item field
+        List<InvoiceItem> result = vertexTaxCalculator.compute(account, invoice, true, Collections.emptyList(), tenantContext);
+        assertEquals(1, result.size());
+        assertNull(result.get(0).getItemDetails());
+        checkTaxItemFields(result.get(0));
+    }
+
+    @Test(groups = "fast")
+    public void testComputeWhenEffectiveTaxRateZero() throws Exception {
+        //given
+        given(invoice.getInvoiceItems()).willReturn(Collections.singletonList(taxableInvoiceItem));
+        given(taxResponse.getData()).willReturn(apiResponseData);
+        final TaxesType taxesType = new TaxesType();
+        taxesType.setCalculatedTax(0d);
+        given(responseLineItem.getTaxes()).willReturn(Collections.singletonList(taxesType));
+
+        //given tax effective rate is present
+        taxesType.setEffectiveRate(0d);
+
+        //then it persisted in item details invoice item field
+        List<InvoiceItem> result = vertexTaxCalculator.compute(account, invoice, true, Collections.emptyList(), tenantContext);
+        assertEquals(0, result.size());
+    }
+
+    @Test(groups = "fast")
+    public void testComputeWhenEffectiveTaxRateZeroButAmountIsNot() throws Exception {
+        //given
+        given(invoice.getInvoiceItems()).willReturn(Collections.singletonList(taxableInvoiceItem));
```

Comment:
this is not a valid usecase

Reply:
this usecase not valid from business perspective and common sense, but i described my thoughts on it [the reply](https://github.com/killbill/killbill-vertex-plugin/pull/42#discussion_r1690516325). 

I created these test case for these 2 reasons:
1. It shows how system behaves in this edge case
2. It shows that we still create tax items even if taxRate is 0 (tests that we rely not on taxRate but on the amount)

Reply:
discussed